mscore/resourceManager.cpp

```diff
@@ -248,6 +262,142 @@ void ResourceManager::displayLanguages()
+                        all_categories << c2;
+                  }
+            QString url_text = "<a href=\"https://musescore.org" + page_url + "\">" + td_list.item(col).toElement().text() + "</a>";
+            // display a clickable hyperlink for each item
+            QLabel* plugin_name = new QLabel(url_text);
+            plugin_name->setTextFormat(Qt::RichText);
+            plugin_name->setTextInteractionFlags(Qt::TextBrowserInteraction);
+            plugin_name->setOpenExternalLinks(true);
+            pluginsTable->setIndexWidget(pluginsTable->model()->index(row, col++), plugin_name);
+            for (; col < 3; col++)
+                  pluginsTable->setItem(row, col, new QTableWidgetItem(td_list.item(col).toElement().text()));
+            if (page_url.isNull()) {
+                  pluginsTable->setItem(row,col,new QTableWidgetItem("No page URL found."));
+                  continue;
+                  }
+
+            QPushButton* install_button = new QPushButton(tr("Install"));
+            // add categories, name and page_url to properties
+            install_button->setProperty("page_url", page_url);
+            install_button->setProperty("categories", category_list);
+            install_button->setProperty("name", name);
+            install_button->setProperty("row", row);
+            QPushButton* uninstall_button = new QPushButton(tr("Uninstall"));
+            uninstall_button->setProperty("page_url", page_url);
+            uninstall_button->setProperty("row", row);
+            uninstall_button->setEnabled(false);
+            connect(uninstall_button, SIGNAL(clicked()), this, SLOT(uninstallPluginPackage()));
+
+            pluginsTable->setIndexWidget(pluginsTable->model()->index(row, col++), install_button);
+            pluginsTable->setIndexWidget(pluginsTable->model()->index(row, col), uninstall_button);
+            refreshPluginButton(row);
+            }
+      categories->insertItems(0,all_categories);
+      }
+
+void ResourceManager::filterPluginList()
+      {
+      const QString category = categories->currentText();
+      const QString& search = lineEdit->text();
+      for (int i = 0; i < pluginsTable->rowCount(); i++) {
+            // locate the "Install" button
+            QPushButton* install = static_cast<QPushButton*>(pluginsTable->indexWidget(pluginsTable->model()->index(i, 3)));
+            const QString& plugin_name = install->property("name").toString();
+            const QStringList& categories = install->property("categories").toStringList();
+            if (plugin_name.contains(search, Qt::CaseInsensitive) && (category == "Any" || categories.contains(category)))
+                  pluginsTable->showRow(i);
+            else
+                  pluginsTable->hideRow(i);
+            }
+      }
+
+void ResourceManager::displayPlugins()
+      {
+      pluginTreeWidget->clear();
+      foreach(const QString &pkg, pluginDescriptionMap.keys()) {
```

Comment:
Please change to `for  (const QString &pkg : pluginDescriptionMap.keys()) {`

Reply:
Resolved in 77701599820befd5e5381af2075d63e6b5b3dd85.